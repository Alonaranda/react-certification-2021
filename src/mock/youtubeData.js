export const videoMock = {
    id: 'id',
    title: 'title',
    description: 'description',
    url: 'url'
}