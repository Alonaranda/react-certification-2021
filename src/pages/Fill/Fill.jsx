import React, { useEffect, useState } from 'react';
import { CardItem } from '../../components/CardItem';
import { getVids } from '../../components/helpers/getVids';
import './Fill.style.css';


const Fill = () => {

  const [videos, setVideos] = useState([]);

  useEffect(() => {
        getVids().then(vids => {
            setVideos(vids)
        })

  }, [])

  return (
    <div>
      <div className="contenedor">
        
        <div className="cards">
            {
                videos.map(vid =>(
                    <CardItem key={vid.id} {...vid}/>
                ))
            }

        </div>

      </div>
    </div>
  );
};

export default Fill;

