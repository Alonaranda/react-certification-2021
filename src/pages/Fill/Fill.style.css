.contenedor {
  margin: auto;
  padding-top: 80px;
  width: 90%;
}

@media (max-width: 480px) {
  .contenedor {
      padding-top: 200px;
  }
  h1{
    margin-top: 30px;
  }
}

h1{
  font-family: monospace;
  text-transform: capitalize;
  text-align: center;
  font-size: 50px;
  margin-top: 5px;
}

.cards {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-evenly;
  margin: 5px 0;
}

.card-item {
  border: 1px solid #ddd;
  border-radius: 10px;
  height: auto;
  width: 300px;
  margin: 15px;
  padding: 15px 15px;
  box-shadow: 6px 9px 16px -1px rgba(0, 0, 0, 0.53); 
  -webkit-box-shadow: 6px 9px 16px -1px rgba(0, 0, 0, 0.53);
  -moz-box-shadow: 6px 9px 16px -1px rgba(0, 0, 0, 0.53);
}

.card-item Link{
  background-color: red
}



.card-item h2 {
  font-size: 1.5em;
  margin: 5px 0;
  text-align: center;
}

.card-item p{
  margin: auto;
  margin-bottom: 5px;
  text-align: justify;
}

.card-item img{
  border-radius: 10px;;
  height: auto;
  width: 100%;
}



@media (min-width: 320px) and (max-width: 768px) {
  .card-item {
    border-radius: 5px;
    height: auto;
    margin: 10px;
    width: 80%;
  }
}
