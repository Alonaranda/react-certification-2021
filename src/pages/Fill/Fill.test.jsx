import React from 'react'
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
//import { render, screen } from "@testing-library/react";
import { Fill }  from './Fill';
//import { CardItem } from '../../components/CardItem';
import { videoMock } from '../../mock/youtubeData';


describe('Test Fill', () => {

    it('should render', () => {
        const wrapper = shallow(<Fill categoria={'Prueba'}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should dont have videos', () => {
        const wrapper = shallow(<Fill categoria={'Prueba'}/>);
        const cardItem = wrapper.find('CardItem');
        expect(cardItem.length).toBe(0);
    });

    it('should have videos',  () => {
        const item = videoMock;
        const result = renderer.create(<Fill categoria={'Prueba'}/>);
        expect(result).toMatchSnapshot();
    });
    
});