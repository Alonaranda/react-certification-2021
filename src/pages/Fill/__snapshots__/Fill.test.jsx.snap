// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Fill should have videos 1`] = `
<div>
  <div
    className="contenedor"
  >
    <h1>
      Loading...
    </h1>
    <div
      className="cards"
    />
  </div>
</div>
`;

exports[`Test Fill should render 1`] = `ShallowWrapper {}`;
